src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -404,27 +413,51 @@ library GPv2Encoding {
                 shr(96, calldataload(encodedInteractions.offset))
             )
 
-            // dataLength = uint24(encodedInteractions[1])
+            // hasValue = bool(encodedInteractions[20])
+            hasValue := iszero(
```

Comment:
is the double iszero necessary given that you are defaulting in the switch below anyways to "has value"

Reply:
It is because we use `if (hasValue)` outside of assembly which requires `hasValue` to be a `bool` which is a `uint1` in Solidity (or else I'm assuming it would be undefined behaviour).

Reply:
Here is the relevant section in the docs: https://docs.soliditylang.org/en/v0.7.6/abi-spec.html?highlight=bool#types